src/client_side_request.h

```diff
@@ -62,14 +62,19 @@ class ClientHttpRequest
     _SQUID_INLINE_ ConnStateData * getConn() const;
     _SQUID_INLINE_ void setConn(ConnStateData *);
 
+    /// initializes request, al->request and al->notes fields
+    void initRequest(HttpRequest *aRequest);
+    /// initializes al->request and al->notes fields
```

Comment:
Same virgin/adapted concern here. Based on the implementation, it is pointless to call initAleRequest multiple times, which implies that it should be called (once) when the virgin request becomes known, but then it is puzzling to see any external calls to this method -- initRequest() should be the only caller.